components/raftstore/src/store/peer_storage.rs

```diff
@@ -1074,6 +1090,21 @@ where
             let rid = self.get_region_id();
             self.engines.raft.gc_entry_cache(rid, idx);
         }
+
+        let mut snap_state = self.snap_state.borrow_mut();
```

Comment:
How about making `cancel_generating_snap` accept an index parameter, so it can be reused here?